operator/src/main/java/org/bf2/operator/operands/KafkaCluster.java

```diff
@@ -815,19 +826,26 @@ private Storage buildKafkaStorage(ManagedKafka managedKafka, Kafka current, long
         long storageBytes = storagePerBroker;
         storageBytes += calculateSafetyMargin(managedKafka, current);
         storageBytes += calculateFormatOverheadFromFormattedSize(managedKafka, storageBytes);
+        int volumeCount = this.configs.getConfig(managedKafka).getKafka().getVolumeCount();
 
-        PersistentClaimStorageBuilder builder = new PersistentClaimStorageBuilder()
-                .withId(JBOD_VOLUME_ID)
-                .withSize(getAdjustedMaxDataRetentionSize(current, storageBytes).getAmount())
-                .withDeleteClaim(DELETE_CLAIM);
+        List<SingleVolumeStorage> volumes = new ArrayList<>(volumeCount);
+
+        for (int jbodVolumeId = 0; jbodVolumeId < volumeCount; jbodVolumeId++) {
+            PersistentClaimStorageBuilder builder = new PersistentClaimStorageBuilder();
+            builder.withId(jbodVolumeId)
+                    .withSize(getAdjustedMaxDataRetentionSize(current, storageBytes, jbodVolumeId).getAmount())
```

Comment:
This assumes we want each volume to have an equal size how valid is that assumption?